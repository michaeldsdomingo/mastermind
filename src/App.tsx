import React, {useState} from 'react';
import {BrowserRouter as Router, Switch, Route} from "react-router-dom";
import ImageSearch from './components/ImageSearch';
import Navbar from './components/Navbar';
import Images from './components/Images';
import 'bootstrap/dist/css/bootstrap.min.css';

import './styles/app.css';
require('dotenv').config();

function App() {
  const [images, setImages] = useState([]);
  const [query, setQuery] = useState("");
  const [isModalShow, setIsModalShow] = useState(false);

  return (
    <div className="App">
      <Router>
        <Switch>
          <Route path="/" exact>
            <div id="main-form">
              <ImageSearch direction="vertical" setImages={setImages} query={query} setQuery={setQuery}/>
            </div>
          </Route>
          <Route path="/test">
            <Navbar setImages={setImages} query={query} setQuery={setQuery}/>
            <div id="test"><div id="test-inner"></div></div>
            {images.length > 1 && <Images images={images} toggleModalShow={setIsModalShow} isModalShow={isModalShow}/>}
          </Route>
        </Switch>
      </Router>
    </div>
  );
}

export default App;
