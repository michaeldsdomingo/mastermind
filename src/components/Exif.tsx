import React from 'react';

type ExifProps = {
    label: string,
    value: any
}

const Exif = ({label, value}: ExifProps) => {
    const formatString = (detail: string) =>  {
        return detail.replace("_", " ").toUpperCase();
    }

    return (
        <div className="exif-container">
            <div className="exif-value purple">{value}</div>
            <div className="exif-label">{formatString(label)}</div>
        </div>
    )
    
}

export default Exif;