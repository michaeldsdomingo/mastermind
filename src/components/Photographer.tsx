import React from 'react';
import DownloadArrow from './DownloadArrow'
// import {Link} from 'react-router-dom'

type PhotographerProps = {
    user: any
}

const Photographer = ({ user }: PhotographerProps) => {
    
    return (
        <div id="photographer-download-container">
            <div id="photographer-container">
                <img  src={user.profile_image.medium} alt="Photographer's profile of self" id="modal-profile-picture"/>
                <div>
                    <div id="photographer-name">{user.name}</div>
                    <a href={user.links.html} target="_blank" id="photographer-username">@{user.username}</a>
                </div>
            </div>
            <button id="download-button">
                {/* <div id="download-arrow"> */}
                    <DownloadArrow />
                {/* </div> */}
                <div id="download">Download</div>
            </button>

        </div>
    )
    
}

export default Photographer