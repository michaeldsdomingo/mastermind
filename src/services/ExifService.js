export const checkExif = (image) => {
    let exifExists = true;
    image.exif && Object.keys(image.exif).forEach((key) => {
      if (!image.exif[key]) exifExists = false;
    })

    return exifExists;
}