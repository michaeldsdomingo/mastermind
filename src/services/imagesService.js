import axios from "axios";

const unsplashEndpoint = `https://api.unsplash.com`;

const getImagesBySearch = (query, page) => {
  let url = `${unsplashEndpoint}/search/photos?client_id=${process.env.REACT_APP_UNSPLASH_ACCESS_KEY}&page=${page}per_page=9`
  if (query) {
    url += `&query=${query}`
  }

  const config = {
    method: "GET",
    url,
    // withCredentials: true,
    crossdomain: true,
    headers: { "Content-Type": "application/json" },
  };
  return axios(config)
}

const getImageById = (photoId) => {
  const config = {
    method: "GET",
    url: `${unsplashEndpoint}/photos/${photoId}?client_id=${process.env.REACT_APP_UNSPLASH_ACCESS_KEY}`,
    // withCredentials: true,
    crossdomain: true,
    headers: { "Content-Type": "application/json" },
  };
  return axios(config)
}

const downloadImage = (url) => {
  const config = {
    method: "GET",
    url: `${url}?client_id=${process.env.REACT_APP_UNSPLASH_ACCESS_KEY}`,
    // withCredentials: true,
    crossdomain: true,
    headers: { "Content-Type": "application/json" },
  };
  return axios(config)
}



export { getImagesBySearch, getImageById, downloadImage }