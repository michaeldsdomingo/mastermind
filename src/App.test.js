import React from "react";
import { shallow, mount } from "enzyme";
import App from "./App";
import Exif from './components/Exif'
import Photographer from './components/Photographer'
import {checkExif} from './services/ExifService'

import toJson from "enzyme-to-json";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { isExportDeclaration } from "typescript";

configure({ adapter: new Adapter() });

it("renders without crashing", () => {
    shallow(<App />);
});

const label="make";
const value="Sony"
  
describe("", () => {
    it("accepts exif props", () => {
        const wrapper = mount(<Exif label={label} value={value} />)
        expect(wrapper.props().label).toEqual(label)
        expect(wrapper.props().value).toEqual(value)
    })
})

const user = {
    name: "Abby Tennisen",
    profile_image: {
        large: 'image-large.com',
        medium: 'image-medium.com',
        small: 'image-small.com',
    },
    username: 'abbyandcamera',
    links: {
        html: 'unsplash.com/abbyandcamera'
    }
}

const downloadURL = 'downloadURL.com'

describe("", () => {
    it("accepts Photographer props", () => {
        const wrapper = mount(<Photographer user={user} downloadURL={downloadURL} />)
        expect(wrapper.props().user).toEqual(user)
    })
})

it("Exif renders correctly", () => {
    const tree = shallow(<Exif label={label} value={value} />);
    expect(toJson(tree)).toMatchSnapshot();
})

it("Photographer renders correctly", () => {
    const tree = shallow(<Photographer user={user} downloadURL={downloadURL} /> )
    expect(toJson(tree)).toMatchSnapshot();
})

describe("Check if Exif exists", () => {
    test("it should return true if none of the keys are null", () => {
        const image = {
            exif: {
                make: "Sony",
                model: "Sony Model 2230",
                aperture: "200"
            }
        }
        expect(checkExif(image)).toEqual(true);
    })
})

describe("Check if Exif exists", () => {
    test("it should return false if any of the keys are null", () => {
        const image = {
            exif: {
                make: "Sony",
                model: null,
                aperture: "200"
            }
        }
        expect(checkExif(image)).toEqual(false);
    })
})