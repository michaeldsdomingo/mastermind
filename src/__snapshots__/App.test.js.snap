// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exif renders correctly 1`] = `
<div
  className="exif-container"
>
  <div
    className="exif-value purple"
  >
    Sony
  </div>
  <div
    className="exif-label"
  >
    MAKE
  </div>
</div>
`;

exports[`Photographer renders correctly 1`] = `
<div
  id="photographer-download-container"
>
  <div
    id="photographer-container"
  >
    <img
      alt="Photographer's profile of self"
      id="modal-profile-picture"
      src="image-medium.com"
    />
    <div>
      <div
        id="photographer-name"
      >
        Abby Tennisen
      </div>
      <a
        href="unsplash.com/abbyandcamera"
        id="photographer-username"
        rel="noopener noreferrer"
        target="_blank"
      >
        @
        abbyandcamera
      </a>
    </div>
  </div>
  <a
    download=""
    href="downloadURL.com?force=true"
    id="download-button"
    rel="nofollow"
  >
    <DownloadArrow />
    <div
      id="download"
    >
      Download
    </div>
  </a>
</div>
`;
